Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/CollectionNode.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    /// <summary>
+    /// Collection node is a data storage class for individual data about an object in the collection.
+    /// </summary>
+    [Serializable]
+    public class CollectionNode
```

Comment:
Should this be a struct?

Reply:
If we end up changing this to a struct we should also move it to the core/definitions/utilities folder

Reply:
I realized I have null checks for the CollectionNode list. I switched it back to a class.

Reply:
Why would a node be null?